microsoft_auth/backends.py

```diff
@@ -160,7 +160,7 @@ def _verify_microsoft_user(self, microsoft_user, data):
             try:
                 # create new Django user from provided data
                 user = User.objects.get(
-                    username=data["preferred_username"][:150]
+                    username=data.get('email', None)
```

Comment:
Why is this line still not preferred_username?

Reply:
Sorry about that, I've realized that preferred_username is also a mutable value (as documented here: https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens).

I'm in the process of reworking this now.

Reply:
@AngellusMortis all ready for you to verify.